### PR TITLE
quick & dirty RP executor

### PR DESCRIPTION
**Pull request by andre-merzky**
As the title says, it's a quick attempt on an RP executor.  You'd need to `pip install radical.pilot` to get it to work, and you'll need to set `RADICAL_PILOT_DBURL` to point to a usable MongoDB.

### CODE REVIEW

**Review by hategan — APPROVED** [review]
I approve.

But this does bring about an interesting issue, which is mostly academic and not meant to hold this PR, that of sprawling dependencies. Ideally, we'd want some solution that is more in-line with the idea that particular users will only use a subset of the connectors. Consequently, we may want a mechanism that allows users to toggle connectors and their dependencies, which may include an actual such mechanism, or having connectors in separate packages/repos.

File: tests/run_local_job.py [hunk 1]
```diff
@@ -1,11 +1,18 @@
...
 from psi.j import Job, JobExecutor, JobSpec
 
 
 if __name__ == '__main__':
+    url = 'local'
+    if len(sys.argv) > 1:
+        url = sys.argv[1]
     job = Job(JobSpec(executable='/bin/date'))
-    exec = JobExecutor.get_instance('local')
+    exec = JobExecutor.get_instance(url)
```
[the remark this answers is not in the document]
Comment (andre-merzky) [reply tests/run_local_job.py:15]: I am sorry that I missed that (out of habit likely), will change.

**Comment by andre-merzky** [comment]
> But this does bring about an interesting issue, which is mostly academic and not meant to hold this PR, that of sprawling dependencies. Ideally, we'd want some solution that is more in-line with the idea that particular users will only use a subset of the connectors. Consequently, we may want a mechanism that allows users to toggle connectors and their dependencies, which may include an actual such mechanism, or having connectors in separate packages/repos.

This implementation has (at the moment) a static plugin mechanism.  One can avoid the dependency issue by allowing 3rd part connector implementors to write their own modules (with their respective dependencies), and to install those modules separately.  The plugins can then be picked by the implementation at runtime, w/o pulling all dependencies into this module.

Let me know if you want some code to demonstrate that.

**Comment by hategan** [comment]
> This implementation has (at the moment) a static plugin mechanism. One can avoid the dependency issue by allowing 3rd part connector implementors to write their own modules (with their respective dependencies), and to install those modules separately. The plugins can then be picked by the implementation at runtime, w/o pulling all dependencies into this module.
> 
> Let me know if you want some code to demonstrate that.

Is this statement a victim of there being two psi-j libraries maybe? What you describe about is indeed an intentional thing with this particular library, so I don't think proof is necessary in this context.

But, because this PR is not about a separately installed package, I think we should have a mechanism that allows selection of connectors that are built into the core library and installation of relevant dependencies. Alternatively, we could simply install all dependencies. In that case, you should update requirements.txt accordingly.

**Comment by andre-merzky** [comment]
> I approve.
> 
> But this does bring about an interesting issue, which is mostly academic and not meant to hold this PR, that of sprawling dependencies. Ideally, we'd want some solution that is more in-line with the idea that particular users will only use a subset of the connectors. Consequently, we may want a mechanism that allows users to toggle connectors and their dependencies, which may include an actual such mechanism, or having connectors in separate packages/repos.

@hategan (and @SteVwonder , @mturilli  I guess): you may want to have a look at [jpsi-python-rp](https://github.com/ExaWorks/jpsi-python-rp).  That repo hosts an RP connector to the [jpsi-python](https://github.com/ExaWorks/jpsi-python) implementation.  The connector is installed separately and with it's own dependencies, and `jpsi-python` will automatically pick it up when installed.  Just saying that this is very much a solvable problem.  I did not do that for the `psi-j-python` stack, but the the same mechanism works here also.

**Pull request merged**